engine/src/function.cpp

```diff
@@ -1235,14 +1236,23 @@ void Function::adjustAttribute(qreal fraction, int attributeIndex)
         return;
 
     //qDebug() << Q_FUNC_INFO << "idx:" << attributeIndex << ", val:" << fraction;
-    m_attributes[attributeIndex].value = CLAMP(fraction, 0.0, 1.0);
+    if(attributeIndex == Function::CrosfaderId){
+        m_attributes[attributeIndex].value = fraction;
+    } else {
+        m_attributes[attributeIndex].value = CLAMP(fraction, 0.0, 1.0);
+    }
     emit attributeChanged(attributeIndex, m_attributes[attributeIndex].value);
 }
 
 void Function::resetAttributes()
 {
-    for (int i = 0; i < m_attributes.count(); i++)
-        m_attributes[i].value = 1.0;
+    for (int i = 0; i < m_attributes.count(); i++){
+        if(!QString::compare(m_attributes[i].name, "CrosfaderId", Qt::CaseInsensitive)){
```

Comment:
This line is the one that broke EFX tests. The "!" should probably be not there.
You effectively set Intensity for all functions to 0 here.